genesyscloud/dependent_consumers/genesyscloud_dependent_consumer_proxy.go

```diff
@@ -0,0 +1,101 @@
+	"github.com/mypurecloud/platform-client-sdk-go/v115/platformclientv2"
+)
+
+type DependentConsumerProxy struct {
+	ClientConfig                   *platformclientv2.Configuration
+	ArchitectApi                   *platformclientv2.ArchitectApi
+	RetrieveDependentConsumersAttr retrieveDependentConsumersFunc
+	GetPooledClientAttr            retrievePooledClientFunc
+}
+
+func (p *DependentConsumerProxy) GetDependentConsumers(ctx context.Context, resourceKeys resourceExporter.ResourceInfo) (resourceExporter.ResourceIDMetaMap, error) {
+	return p.RetrieveDependentConsumersAttr(ctx, p, resourceKeys)
+}
+
+func (p *DependentConsumerProxy) GetAllWithPooledClient(method gcloud.GetAllConfigFunc) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	return p.GetPooledClientAttr(method)
+}
+
+type retrieveDependentConsumersFunc func(ctx context.Context, p *DependentConsumerProxy, resourceKeys resourceExporter.ResourceInfo) (resourceExporter.ResourceIDMetaMap, error)
+type retrievePooledClientFunc func(method gcloud.GetAllConfigFunc) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics)
+
+var InternalProxy *DependentConsumerProxy
+
+// getDependentConsumerProxy acts as a singleton to for the InternalProxy.
+func GetDependentConsumerProxy(ClientConfig *platformclientv2.Configuration) *DependentConsumerProxy {
+	return newDependentConsumerProxy(ClientConfig)
+}
+
+// newDependentConsumerProxy initializes the ruleset proxy with all of the data needed to communicate with Genesys Cloud
+func newDependentConsumerProxy(ClientConfig *platformclientv2.Configuration) *DependentConsumerProxy {
+	if InternalProxy == nil {
+		InternalProxy = &DependentConsumerProxy{
+			GetPooledClientAttr: retrievePooledClientFn,
+		}
+	}
+
+	if ClientConfig != nil {
+		api := platformclientv2.NewArchitectApiWithConfig(ClientConfig)
+		InternalProxy.ClientConfig = ClientConfig
+		InternalProxy.ArchitectApi = api
+		InternalProxy.RetrieveDependentConsumersAttr = retrieveDependentConsumersFn
+	}
+
+	return InternalProxy
+}
+
+func retrievePooledClientFn(method gcloud.GetAllConfigFunc) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resourcefunc := gcloud.GetAllWithPooledClient(method)
+	ctx, _ := context.WithCancel(context.Background())
+	resources, err := resourcefunc(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return resources, err
+}
+
+func retrieveDependentConsumersFn(ctx context.Context, p *DependentConsumerProxy, resourceKeys resourceExporter.ResourceInfo) (resourceExporter.ResourceIDMetaMap, error) {
+	resourceKey := resourceKeys.State.ID
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	if resourceKeys.Type == "genesyscloud_flow" {
+		dependentConsumerMap := SetDependentObjectMaps()
+		data, _, err := p.ArchitectApi.GetFlow(resourceKey, false)
+		if err != nil {
+			log.Printf("Error calling GetFlow: %v\n", err)
+		}
+		if data != nil && data.PublishedVersion != nil && data.PublishedVersion.Id != nil {
```

Comment:
I've noticed that there are some flows that don't have a PublishedVersion ... do we know why this is the case and how to pass in the version in this case?

Reply:
You might be working on a flow the first time and never publish it.